production/loki-mixin/alerts.libsonnet

```diff
@@ -70,6 +70,25 @@
               |||, 'cluster', $._config.per_cluster_label),
             },
           },
+          {
+            // Alert if the compactor has not successfully run compaction in the last 24h.
+            alert: 'LokiCompactorHasNotSuccessfullyRunCompaction',
+            expr: |||
+              # The "last successful run" metric is updated even if the compactor owns no tenants,
+              # so this alert correctly doesn't fire if compactor has nothing to do.
+              (time() - loki_compactor_apply_retention_last_successful_run_timestamp_seconds > 60 * 60 * 24)
```

Comment:
nit: this could be flaky if the very first compaction run takes longer than an hour,
Internally we run two variants of this:

since-last-iteration
```
(
time() -
  (loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds{} > 0)
      > THRESHOLD
)
```

since start-up
```
max(max_over_time(loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds{}[THRESHOLD])) by (cluster, namespace) == 0
```

Reply:
I'll add the second one but could you explain why this is flaky?

Reply:
if the very first compaction run takes longer than an hour, the metric value would be 0 for the entire evaluation period with the alert expression evaluating to true

Reply:
Thanks :)

Reply:
np. it could page on restarts as well :)

Reply:
Hey @ashwanthgoli I think I should have addressed all your comments :)